drv/ksz8463/src/lib.rs

```diff
@@ -0,0 +1,149 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+#![no_std]
+
+use core::convert::TryInto;
+
+use drv_spi_api::{SpiDevice, SpiError};
+use drv_stm32h7_gpio_api as gpio_api;
+use gpio_api::{Gpio, OutputType, PinSet, Pull, Speed};
+use ringbuf::*;
+use userlib::{hl::sleep_for, task_slot};
+
+task_slot!(GPIO, gpio_driver);
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+enum Trace {
+    None,
+    Read(Register, u16),
+    Write(Register, u16),
+    Id(u16),
+}
+ringbuf!(Trace, 16, Trace::None);
```

Comment:
Decided not to do anything about, since it would be a weird choice to have two tasks that both drive KSZ chips (we have no hardware with > 1 KSZ chip, and no plans to build such a thing).